python/lsst/obs/base/fitsRawFormatterBase.py

```diff
@@ -118,21 +125,82 @@ def makeVisitInfo(self, metadata):
         """
         raise NotImplementedError("Must be implemented by subclasses.")
 
-    def makeWcs(self, metadata):
-        """Construct a SkyWcs from metadata.
+    @abstractmethod
+    def getDetector(self, id):
+        """Return the detector that acquired this raw exposure.
 
         Parameters
         ----------
+        id : `int`
+            The identifying number of the detector to get.
+
+        Returns
+        -------
+        detector : `~lsst.afw.cameraGeom.Detector`
+            The detector associated with that ``id``.
+        """
+        raise NotImplementedError("Must be implemented by subclasses.")
+
+    def makeWcs(self, visitInfo, detector, metadata):
+        """Construct a SkyWcs from the visitInfo and detector, or metadata
+        of an exposure.
```

Comment:
When I first read this I thought it meant that sometimes `visitInfo` and `detector` would be provided, and sometimes `metadata` would be provided, and the implementation had to handle both cases.  Please clarify (perhaps when you fill in the parameter descriptions below).

Reply:
:laughing: I totally botched those `Descriptions`, didn't I.